src/lxml/html/clean.py

```diff
@@ -223,6 +223,14 @@ def __init__(self, **kw):
         if self.inline_style is None and 'inline_style' not in kw:
             self.inline_style = self.style
 
+        assert not all([kw.get("allow_tags"), kw.get("remove_unknown_tags")]), \
```

Comment:
It currently raises a `ValueError`, which generally seems more appropriate, but is also better because assertions can be disabled. (And just merge it into the case below.)

Reply:
Done.